Base/splitter.h

```diff
@@ -0,0 +1,42 @@
+#ifndef SPLITTER_H__
+#define SPLITTER_H__
+
+struct SplitterRecord
+{
+  double threshold;       // Threshold to split at.
+  double improvement;     // Impurity improvement given parent node.
+  double impurity_left;   // Impurity of the left split.
+  double impurity_right;  // Impurity of the right split.
+};
+
+class SplitterInterface{
+  public:
+    virtual ~SplitterInterface() { }
+
+    /**
+     * Initialize the splitter (Used by Tree)
```

Comment:
Use a dot at the end of each line, otherwise we won't know if you have finished saying what you wanted.